opentelemetry-sdk/src/logs/growable_array.rs

```diff
@@ -0,0 +1,370 @@
+/// The default max capacity for the stack portion of `GrowableArray`.
+const DEFAULT_MAX_STACK_CAPACITY: usize = 10;
```

Comment:
```suggestion
const DEFAULT_MAX_STACK_CAPACITY: usize = 5;
```

Add a link to the study in Golang : https://go.dev/blog/slog so we know where is 5 coming from.

Reply:
This is handled separately for LogRecord in record.rs. Didn't want to impose the recommended size of LogRecord attributes here:
```rust
// According to a Go-specific study mentioned on https://go.dev/blog/slog,
// up to 5 attributes is the most common case.
const PREALLOCATED_ATTRIBUTE_CAPACITY: usize = 5;

/// A vector of `Option<(Key, AnyValue)>` with default capacity.
pub(crate) type AttributesGrowableArray =
    GrowableArray<Option<(Key, AnyValue)>, PREALLOCATED_ATTRIBUTE_CAPACITY>;
```